app/lib/actions/katello/repository/clone_to_version.rb

```diff
@@ -22,21 +22,31 @@ def plan(repositories, content_view_version, options = {})
               fail "Cannot clone metadata if more than one repository" if repositories.count > 1
               plan_action(Repository::CloneYumMetadata, repositories[0], new_repository)
             else
-              repositories.each do |repository|
-                if new_repository.yum?
-                  plan_action(Repository::CloneYumContent, repository, new_repository, filters, :purge_empty_units => !incremental,
-                              :generate_metadata => !incremental, :index_content => !incremental, :simple_clone => incremental, :rpm_filenames => rpm_filenames)
-                elsif new_repository.deb?
-                  plan_action(Repository::CloneDebContent, repository, new_repository, filters, !incremental,
-                              :generate_metadata => !incremental, :index_content => !incremental, :simple_clone => incremental)
-                elsif new_repository.docker?
-                  plan_action(Repository::CloneDockerContent, repository, new_repository, filters)
-                elsif new_repository.ostree?
-                  plan_action(Repository::CloneOstreeContent, repository, new_repository)
-                elsif new_repository.file?
-                  plan_action(Repository::CloneFileContent, repository, new_repository)
-                end
-              end
+              clone_repository_content(repositories, new_repository, filters, incremental, rpm_filenames)
+            end
+
+            plan_action(Katello::Repository::MetadataGenerate, new_repository) if repositories.length > 1
+          end
+        end
+
+        def clone_repository_content(repositories, new_repository, filters, incremental, rpm_filenames)
+          repositories.each do |repository|
+            if new_repository.yum?
+              # If there is more than one repository passed here, that means that there are duplicate repos in a composite content view.
+              # We skip generating metadata in this case and generate it later to prevent conflicting data, such as filters.
```

Comment:
if the metadata generation gets skipped here, where does it happen later on?

Reply:
In this step `plan_action(Katello::Repository::MetadataGenerate, new_repository) if repositories.length > 1`

https://github.com/Katello/katello/pull/7836/files#diff-516603408c98ec56e67c629e9de94587R28

Reply:
oops, didnt see that even though it was right on the screen :eyeglasses: 

it's correct that in the case of incremental, no metadata is regenerated? The `skip_metadata` has two checks, but the regen at the end only validates the latter check

Reply:
hrmmm... good point. Do we do incremental versions for composite content views?

Reply:
I think so, at least based on the text for the param in the api:

`Content View Version Ids to perform an incremental update on.  May contain composites as well as one or more components to update.`

Reply:
I'm not sure if there is a better way to display the logic, this was the most readable way I could figure out, but I'm open to suggestions :open_file_folder: 

The original functionality should be working as is. Here are some simplified scenarios

``` ruby
# incremental publish and normal CV
2.5.1 :002 > incremental = true; repositories = ['foo']
 => ["foo"]
# skip metadata 
2.5.1 :003 > incremental || repositories.length > 1
 => true 

# incremental and CVV
2.5.1 :004 > incremental = true; repositories = ['foo', 'bar']
 => ["foo", "bar"] 
2.5.1 :005 > incremental || repositories.length > 1
 => true 

# normal publish and CVV
2.5.1 :006 > incremental = false; repositories = ['foo', 'bar']
 => ["foo", "bar"] 
2.5.1 :007 > incremental || repositories.length > 1
 => true 

# normal publish and CV
2.5.1 :008 > incremental = false; repositories = ['foo']
 => ["foo"] 
# here we don't skip metadata regenerating in the task
2.5.1 :009 > incremental || repositories.length > 1
 => false 
```

If any CV publish is incremental, we are still skipping metadata, the addition is we are skipping metadata if its a CVV with duplicate repos as well